File: app/src/main/java/com/example/tanjinc/myapplication/fragment/gridpage/GridContract.java
package com.example.tanjinc.myapplication.fragment.gridpage;

import com.example.tanjinc.myapplication.bean.BaseBean;

import java.util.List;

public class GridContract {

    public interface View {
        void showDataList(List<? extends BaseBean> dataList, boolean isAppend);

        void setPresenter(Presenter presenter);
    }

    public interface Presenter {
        void loadData();       //加载数据

        void loadMoreData();   //加载更多

        void refresh();         //刷新
    }
}

File: app/src/main/java/com/example/tanjinc/myapplication/fragment/gridpage/GridContract.kt
package com.example.tanjinc.myapplication.fragment.gridpage

import com.example.tanjinc.myapplication.bean.BaseBean

class GridContract {

    interface View {
        fun showDataList(dataList: List<BaseBean>, isAppend: Boolean)

        fun setPresenter(presenter: Presenter)
    }

    interface Presenter {
        fun loadData()        //加载数据

        fun loadMoreData()    //加载更多

        fun refresh()          //刷新
    }
}
